pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -0,0 +1,963 @@
+package alerting
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/notifier/channels"
+	"github.com/grafana/grafana/pkg/tests/testinfra"
+)
+
+// alertmanagerConfig has the config for all the notification channels
+// that we want to test. It is recommended to use different URL for each
+// channel and have 1 route per channel.
+const alertmanagerConfig = `
+{
+  "alertmanager_config": {
+    "route": {
+      "receiver": "slack_recv1",
+      "group_wait": "0s",
```

Comment:
```suggestion
      "group_wait": "0s", // no group_wait time means the notification is sent as soon as it is received.
```

Reply:
Can't have comments in JSON :) so I will add this as part of the variable comment